src/init.cpp

```diff
@@ -1562,7 +1581,11 @@ bool AppInitMain()
     }
 
     // ********************************************************* Step 9: load wallet
-    if (!g_wallet_init_interface.Open()) return false;
+    for (const auto& client : interfaces.chain_clients) {
```

Comment:
Could use `std::any_of`? :-)

Reply:
@practicalswift just a meta-note, could you please put all your comments in one review rather than putting them all as separate comments?

Reply:
@MeshCollider Sure!

Reply:
Also, I'd prefer if we didn't leave style feedback at all.

Of course it could use std::any_of, or it could use a while loop, or do while loop, or... It is just a different color up to the author to choose and for everyone else to accept.

Reply:
@MarcoFalke I should have made clear that my comment was a "nit" :-)

The comparison you're making isn't fair: no one would suggest changing this to a `while` loop, but striving to use STL algorithms were possible is what C++ Core Guidelines (with editors Bjarne Stroustrup and Herb Sutter) recommends:

> ES.1: Prefer the standard library to other libraries and to “handcrafted code”
>
> Code using a library can be much easier to write than code working directly with language features, much shorter, tend to be of a higher level of abstraction, and the library code is presumably already tested. The ISO C++ Standard Library is among the most widely known and best tested libraries. It is available as part of all C++ Implementations.
> 
> https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es1-prefer-the-standard-library-to-other-libraries-and-to-handcrafted-code

Reply:
cmd.to

Reply:
> Could use std::any_of? :-)

This I think it would actually have to use std::all_of

Reply:
re: https://github.com/bitcoin/bitcoin/pull/14437#discussion_r232173026

> Could use std::any_of? :-)

This PR is already merged so I won't change this here. But it's an interesting suggestion. I might prefer `all_of` over `any_of` since the bool represents success, and success only happens if _all_ calls succeed.

Note: one drawback to using any/all in a context like this is it makes behavior non-deterministic, because the standard doesn't guarantee what order the calls will run in or whether calls will be skipped due to short-circuiting: https://stackoverflow.com/questions/47692180/is-stdany-of-required-to-follow-short-circuit-logic